neasqc_wp61/models/quantum/embedder.py

```diff
@@ -0,0 +1,322 @@
+"""
+Embedder
+============
+Module containing the base class for transforming sentences in a dataset into embeddings
+"""
+
+from abc import ABC, abstractmethod
+
+import pandas as pd
+from pandas.core.api import DataFrame as DataFrame
+from transformers import BertTokenizer, BertModel
+import torch
+import numpy as np
+import fasttext as ft
+import fasttext.util as ftu
+
+
+class Embedder(ABC):
+    """
+    Base class for the generation of embeddings for the sentences
+    comprising a dataset.
+    """
+
+    def __init__(self, dataset: DataFrame, sentence_or_word: bool = True) -> None:
+        """
+        Initialises the embedder class
+
+        Parameters
+        ----------
+        dataset : pd.DataFrame
+            Pandas dataset. Each row of the dataset correspods to a
+            sentence.
+        sentence_or_word : bool
+            A bool controllig whether we want to produce a sentence
+            embedding or word embedding vector for each sentence in the
+            dataset. True is for sentence embedding, False is for word
+            embedding.
+        """
+
+        self.dataset = dataset
+        self.sentence_or_word = sentence_or_word
+
+    @abstractmethod
+    def compute_embeddings(self) -> DataFrame:
+        """
+        Creates embeddings for sentences in a dataset
+
+        Returns
+        -------
+        pd.DataFrame
+            A pandas dataframe with the same structure as the original
+            dataset, but with an additional column containing the
+            embeddings corresponding to the sentence contained in each
+            row.
+        """
+        pass
+
+    @abstractmethod
+    def save_embedding_dataset(self, path: str, filename: str) -> None:
+        """
+        Saves the dataset containing the embeddings as a TSV file.
+
+        Parameters
+        ----------
+        path : str
+            The path to the location where the user wishes to save the
+            generated dataset containing the embeddings (it should not
+            include a / at the end).
+        filename: str
+            The name with which the generated dataset containing the
+            embeddings will be saved (it should not include .csv or any
+            other file extension).
+        """
+        self.dataset.to_csv(path + "/" + filename + ".tsv", index=False, sep="\t")
+
+
+class Bert(Embedder):
+    """
+    Class for generating BERT embeddings
+    """
+
+    def __init__(
+        self,
+        dataset: DataFrame,
+        sentence_or_word: bool = True,
+        cased_or_uncased: bool = False,
```

Comment:
same remark as parent class

Reply:
Addressed - would you suggest the same for cased_or_uncased variable, i.e. just changed it for something like cased/casing?

Reply:
yep!